openlibrary/plugins/openlibrary/api.py

```diff
@@ -373,3 +374,17 @@ def GET(self):
                     metadata['ocaid'] = ed.ocaid
 
         return json.dumps(metadata)
+
+
+class observations(delegate.page):
+    path = "/observations"
+    encoding = "json"
+
+    def POST(self):
+        user = accounts.get_current_user()
+        account = OpenLibraryAccount.get_by_email(user.email)
+        s3_keys = web.ctx.site.store.get(account._key).get('s3_keys')
+
+        response = post_observation(web.data(), s3_keys)
+
+        return delegate.RawText(response)
```

Comment:
A bit more protection if e.g. patron tries to perform post while not logged in (don't try to POST)
```suggestion
        user = accounts.get_current_user()
        if user:
            account = OpenLibraryAccount.get_by_email(user.email)
            s3_keys = web.ctx.site.store.get(account._key).get('s3_keys')
            if s3_keys:
                response = post_observation(web.data(), s3_keys)
                return delegate.RawText(response)
```

Reply:
Done.